docs/rules/v-on-handler-style.md

```diff
@@ -46,57 +46,100 @@ This rule aims to enforce a consistent style in `v-on` event handlers:
 ```json
 {
   "vue/v-on-handler-style": ["error",
-    [
-      "method",
-      "inline-function"
-    ],
+    ["method", "inline-function"], // ["method", "inline-function"] | ["method", "inline"] | "inline-function" | "inline"
     {
       "ignoreIncludesComment": false
     }
   ]
 }
 ```
 
-- First option ... An array of names of allowed styles. Default is `["method", "inline-function"]`.
-  - `"method"` ... Allow handlers by method binding. e.g. `v-on:click="handler"`
-  - `"inline"` ... Allow inline handlers. e.g. `v-on:click="handler()"`  
-                  Even if this is not specified, writing styles that cannot be converted to other allowed styles are allowed.
+- First option ... Specifies the name of an allowed style. Default is `["method", "inline-function"]`.
+  - `["method", "inline-function"]` ... Allow handlers by method binding. e.g. `v-on:click="handler"`. Allow inline functions where method handlers cannot be used. e.g. `v-on:click="() => handler(listItem)"`.
+  - `["method", "inline"]` ... Allow handlers by method binding. e.g. `v-on:click="handler"`. Allow inline handlers where method handlers cannot be used. e.g. `v-on:click="handler(listItem)"`.
   - `"inline-function"` ... Allow inline functions. e.g. `v-on:click="() => handler()"`
+  - `"inline"` ... Allow inline handlers. e.g. `v-on:click="handler()"`
 - Second option
-  - `ignoreIncludesComment` ... If `true`, do not report expressions containing comments. This option takes effect if `"method"` is allowed. Default is `false`.
+  - `ignoreIncludesComment` ... If `true`, do not report expressions containing comments to be converted to method handlers. However, it will be reported if the second style is not used. This option takes effect if `"method"` is allowed. Default is `false`.
 
-### `["method"]`
+### `["method", "inline-function"]` (Default)
 
-<eslint-code-block fix :rules="{'vue/v-on-handler-style': ['error', ['method']]}">
+<eslint-code-block fix :rules="{'vue/v-on-handler-style': ['error', ['method', 'inline-function']]}">
 
 ```vue
 <template>
   <!-- ✓ GOOD -->
   <button v-on:click="handler" />
   <template v-for="e in list">
-    <button @click="e" />
-    <button @click="e" />
+    <button v-on:click="e" />
+    <button v-on:click="() => handler(e)" />
   </template>
 
   <!-- ✗ BAD -->
   <button v-on:click="handler()" />
+  <button v-on:click="count++" />
   <button v-on:click="() => handler()" />
+  <button v-on:click="() => count++" />
   <button v-on:click="(a, b) => handler(a, b)" />
   <template v-for="e in list">
-    <button @click="e()" />
-    <button @click="() => e()" />
+    <button v-on:click="e()" />
+    <button v-on:click="() => e()" />
+    <button v-on:click="handler(e)" /> <!-- Can use inline function -->
```

Comment:
This comment is confusing to me; the other lines also have better alternatives but no comment. So I'd remove the comment completely.